inffast.c

```diff
@@ -12,6 +12,14 @@
 #include "functable.h"
 
 
+/* Behave like chunkcopy, but avoid writing beyond of legal output. */
+static inline uint8_t* chunkcopy_safe(uint8_t *out, uint8_t const *from, unsigned len, uint8_t *safe) {
```

Comment:
I would assume this should perform much better.

Reply:
Yep, it definitely seems to by my measurements on x86 and aarch64 at lest.